docs/integrating-with-unity/defining-states.mdx

```diff
@@ -0,0 +1,334 @@
+| `tag`         | `string`     | The tag of this GameObject as seen in the editor, can be null if GameObject is untagged.                                                                                                          |
+| `scene`       | `string`     | The name of the scene this GameObject is part of.                                                                                                                                                 |
+| `pathInScene` | `string`     | The full path to this GameObject in its scene.                                                                                                                                                    |
+| `position`    | `Vector3`    | The position of this GameObject in world space.                                                                                                                                                   |
+| `rotation`    | `Quaternion` | The rotation of this GameObject in world space.                                                                                                                                                   |
+| `isPlayer`    | `boolean`    | If true, this GameObject has a MonoBehaviour representing a player controlled object. That player may or may not be driven by a bot.  (See the [RGStateType] section below for more information.) |
+| `clientId`    | `long?`      | The id of the client that owns/controls this GameObject if it represents a bot.  This field can be null.                                                                                          |
+
+
+## 2 - `[RGStateType]` Class Level Attribute
+By adding the `[RGStateType]` attribute to any MonoBehaviour class in your project, ALL GameObjects with that MonoBehaviour attached will be included in the state.
+
+Any MonoBehaviour using this attribute will be included in the state under a property named the same as the Behaviour's class name or the `typeName` specified by this attribute.
+
+When this attribute is used in a class, a new source file will be autogenerated by Regression Games to represent this entity in the state.
+If this class includes `[RGState]` attributes on fields, properties, or methods, then only those will be included in the state.
+Otherwise, all public fields, properties, and zero parameter methods with return values will be included.
+
+### `[RGStateType]` Example 1
+This example shows using `[RGStateType]` with `isPlayer` set to `true` to indicate that GameObjects with this MonoBehaviour should be considered players.
+
+Source File: KingState.cs
+```cs
+[RGStateType(isPlayer:true)]
+public class KingState : MonoBehaviour
+{
+    public string CharacterType()
+    {
+        return "King";
+    }
+
+    public int Health()
+    {
+        return 100;
+    }
+}
+```
+
+Generated File: Generated_RGStateEntity_KingState.cs
```

Comment:
Again, not sure if we need to include this, or maybe we provide a link to a separate page? Thinking about scanability of the page as well